packages/client-utils/src/vstorage.js

```diff
@@ -1,30 +1,66 @@
+import { decodeBase64 } from '@endo/base64';
+import { encodeHex } from '@agoric/internal/src/hex.js';
+import { StreamCellShape } from '@agoric/internal/src/lib-chainStorage.js';
+import { mustMatch } from '@agoric/internal';
 import {
   QueryChildrenRequest,
   QueryChildrenResponse,
   QueryDataRequest,
   QueryDataResponse,
 } from '@agoric/cosmic-proto/agoric/vstorage/query.js';
-import { decodeBase64 } from '@endo/base64';
-import { encodeHex } from '@agoric/internal/src/hex.js';
 
 /**
+ * @import {AbciQueryResponse} from '@cosmjs/tendermint-rpc';
+ * @import {JsonSafe} from '@agoric/cosmic-proto';
+ * @import {StreamCell} from '@agoric/internal/src/lib-chainStorage.js';
  * @import {MinimalNetworkConfig} from './network-config.js';
  */
 
+/**
+ * @param {'data' | 'children'} kind
+ */
+const getCodecKey = kind => {
+  switch (kind) {
+    case 'children':
+      return '/agoric.vstorage.Query/Children';
+    case 'data':
+      return '/agoric.vstorage.Query/Data';
+    default:
+      throw Error(`invalid codec kind ${kind}`);
+  }
+};
+
+// TODO move down to cosmic-proto, probably generated with Telescope
 const codecs = {
-  children: {
-    rpc: '/agoric.vstorage.Query/Children',
+  '/agoric.vstorage.Query/Children': {
```

Comment:
Mea culpa.  This should probably not have a prototype of `Object.prototype`:
```suggestion
  __proto__: null,
  '/agoric.vstorage.Query/Children': {
```